rfd/0186-optionally-require-reason-for-access-request.md

```diff
@@ -0,0 +1,215 @@
+---
+authors: Pawel Kopiczko (pawel.kopiczko@goteleport.com)
+state: draft
+---
+
+# RFD 186 - Optionally require reason for access request
+
+
+## Required Approvers
+
+* Engineering: @r0mant && @smallinsky
+* Product: @klizhentas && @xinding33
+
+
+## What
+
+Add the ability to set a policy to make a reason required for an access request
+to be submitted
+([#20164](https://github.com/gravitational/teleport/issues/20164)).
+
+
+## Why
+
+Improves user experience for environments with audit/compliance policies that
+require a reason. Adding the ability to mark a reason as required would allow
+both tsh and the web UI to prompt for a reason without the end user needing to
+remember that it's required and as a result reduce the number of invalid access
+requests that need to be rejected.
+
+
+## Details
```

Comment:
Every RFD must start with exploring UX and user scenarios. Before jumping into details, can you please have a UX section that describes scenarios we want to support. For example:

- Requiring a reason for any access requests.
- Requiring a reason for some roles but not others.
- Requiring a reason for some resources but not others.